lib/crawly/spider.ex

```diff
@@ -18,4 +22,7 @@ defmodule Crawly.Spider do
   @callback parse_item(response :: HTTPoison.Response.t()) ::
   Crawly.ParsedItem.t()
 
+  @callback custom_settings() ::  Map.t()
```

Comment:
I think a keyword list would be more appropriate for this, since it matches how settings are declared in config.exs

Reply:
+1. I have also improved the typespec a bit.